pkg/expr/transform.go

```diff
@@ -72,13 +79,30 @@ type Request struct {
 type Query struct {
 	RefID         string
 	TimeRange     TimeRange
-	DatasourceUID string
+	DatasourceUID string        // deprecated, value -100 when expressions
+	Datasource    DataSourceRef `json:"datasource"`
```

Comment:
@kylebrandt do we need a special object here?  can it be the same DataSource object that already exists on q?

Reply:
I believe new alerting use this, so that would have to change.